python/tchannel/tornado/tchannel.py

```diff
@@ -99,15 +102,35 @@ def __init__(self, hostport=None, process_name=None, known_peers=None):
             sys.argv[0], os.getpid()
         )
 
+        self.name = name
+
         # register event hooks
         self.event_emitter = EventEmitter()
         self.hooks = EventRegistrar(self.event_emitter)
-        self.hooks.register(ZipkinTraceHook(tchannel=self))
 
         if known_peers:
             for peer_hostport in known_peers:
                 self.peers.add(peer_hostport)
 
+    def advertise(self, router, name=None):
+        """Advertise the given TChannel to Hyperbahn.
+
+        This informs Hyperbahn that the given client/service is using TChannel
+        at a fixed rate.
+
+        It also tells the TChannel about the given Hyperbahn routers.
+
+        :param routers:
+            Seed list of addresses of Hyperbahn routers
+        :param name:
+            Name to be register on the hyperbahn.
+        :returns:
+            A future that resolves to the remote server's response after
+            the first advertise is successful.
```

Comment:
it resolves to the remote server's response after the first advertise finished. it may or may not have been successful. just say "after the first advertise finishes"